src/oic/oic/provider.py

```diff
@@ -2049,3 +1883,449 @@ def get_by_sub_and_(self, sub: str, key: str, val: Any) -> Optional[str]:
+            else:
+                uid = self.sdb.get_uid_by_sub(sub)
+
+            if not client_id:
+                if len(id_token_hint["aud"]) == 1:
+                    client_id = id_token_hint["aud"][0]
+                else:
+                    client_id = id_token_hint["azp"]
+
+            sids = session_get(self.sdb, "sub", sub)
+
+            for sid in sids:
+                if self.sdb[sid]["client_id"] == client_id:
+                    break
+
+        if not client_id:
+            return error_response("invalid_request", "Could not find client ID")
+        if client_id not in self.cdb:
+            return error_response("invalid_request", "Unknown client")
+
+        redirect_uri = None
+        if "post_logout_redirect_uri" in esr:
+            redirect_uri = self.verify_post_logout_redirect_uri(esr, client_id)
+            if not redirect_uri:
+                msg = "Post logout redirect URI verification failed!"
+                return error_response("invalid_request", msg)
+        else:  # If only one registered use that one
+            try:
+                _ruri = self.cdb[client_id]["post_logout_redirect_uris"]
+            except KeyError:
+                msg = "Missing post_logout_redirect_uri"
+                return error_response("invalid_request", msg)
+
+            if len(_ruri) == 1:
+                _base, _query = _ruri[0]
+                if _query:
+                    query_string = urlencode(
+                        [(key, v) for key in _query for v in _query[key]]
+                    )
+                    redirect_uri = "%s?%s" % (_base, query_string)
+                else:
+                    redirect_uri = _base
+
+        # redirect user to OP logout verification page
+        payload = {
+            "uid": uid,
+            "client_id": client_id,
+            "redirect_uri": redirect_uri,
+            "sid": sid,
+        }
+        if "state" in esr:
+            payload["state"] = esr["state"]
+
+        if self.events:
+            self.events.store("object args", "{}".format(payload))
+
+        # From me to me
+        _jws = JWT(
+            self.keyjar, iss=self.name, lifetime=86400, sign_alg=self.signing_alg
+        )
+        sjwt = _jws.pack(aud=[self.name], **payload)
+
+        location = "{}?{}".format(self.logout_verify_url, urlencode({"sjwt": sjwt}))
+        return SeeOther(location)
+
+    def unpack_signed_jwt(self, sjwt: str):
+        """Will unpack a signed JWT."""
+        verifier = JWT(self.keyjar)
+        try:
+            return verifier.unpack(sjwt)
+        except Exception as err:
+            raise ValueError(err)
+
+    def do_verified_logout(
+        self, sid: str, client_id: str, alla: bool = False, **kwargs
+    ) -> Dict:
+        """
+        Perform back channel logout and prepares the information needed for front channel logout.
+
+        :param sid: Session ID
+        :param client_id:
+        :param alla:
```

Comment:
Have no idea what this stands for... Either make it more descritptive, or add the docstring.